CMakeLists.txt

```diff
@@ -17,6 +17,8 @@
 #
 
 cmake_minimum_required(VERSION 2.6)
+cmake_policy(SET CMP0011 NEW)
+cmake_policy(SET CMP0012 NEW)
```

Comment:
Please add comments here explaining why we need that as in the commit message. Otherwise CMP0012 is not directly related to FindPython3 and FindPython2 thus it would be nice to explain the link, maybe just point to the explanation of what got broken in CMake 3.18.2? Finally, do we know if we will be able to get rid of that at some point? What are the conditions to be fulfilled ?

Reply:
Will do. What got broken in 3.18.2 was that the policy defaulted to OLD, which is what broke FindPython* for us. In 3.18.3 it should default to NEW, however there is no danger in leaving both of these in place really, as these are both policies we have to have set.

Reply:
Thus is it the case that we will have those two cmake_policy calls just to support CMake 3.18.2?

Reply:
No, CMP0011 will be required for > 3.18, otherwise we will incur warnings when building with CMake, which is undesirable.

CMP0012 *may* not be required after 3.18.3, I am not sure yet, but it won't hurt either way. Relying on defaults for polices tends to get you warnings, and sometimes broken.

Without CMP0012 we are currently broken with the latest version of cmake, which is shipped in most rolling distros, unfortunately. I am not sure when the next version of CMake is due, but we already have reports from other projects which use MbedTLS of breakage, thus should really patch this, even if only for now.

Reply:
I understand the necessity of this patch and thanks for it, I just try to understand clearly for what cmake version this is needed or not. Thus for CMP0012:
. before 3.18.2, not needed
. needed for 3.18.2
. shouldn't be necessary after 3.18.3 but not sure yet.
Is that correct?

Reply:
Sorry, I misunderstood where you were headed with the comment.

The confusion comes because I don't know exactly what the issue was, only that either setting this or a max version fixed it. The max version fixed it by setting policies anyway, so this seemed the better (and more long lasting) fix.

According to https://cmake.org/cmake/help/latest/policy/CMP0012.html (these links will be added as comments)
For versions prior to 2.8 this will be ignored
For version 3.18.2 the default was set to OLD, which somehow directly or indirectly broke FindPython* by LOCATION. Therefore this directive is required to fix that.
For version 3.18.3 and beyond, it will still default to OLD, but there will now be a deprecation warning which we would need to fix anyway by specifying NEW. Whether or not setting it to OLD would break FindPython* is unknown, I would hope not, but OLD is still deprecated, so we should not be using it.

There does appear to be a number of changes requiring policies to be set recently. We have had policy statements before, and, it would appear removed them, but this can only be done once there is no OLD behaviour left for the policy.

Reply:
ok thanks for this extra information, I will wait for the comments now.

Reply:
I have tested this with cmake 3.17.4, 3.18.2, 3.18.3, and 3.18.4, and it fixes several issues with several of those versions.